test/docker-e2e/e2e_upgrade_test.go

```diff
@@ -0,0 +1,199 @@
+	s.Require().Equal(strings.TrimPrefix(targetVersion, "v"), abciInfo.Response.GetVersion(), "version mismatch")
+}
+
+// TestCelestiaAppMajorUpgrade tests a major upgrade from v4.0.7-mocha to a commit hash which has v5
+// using the signaling mechanism.
+func (s *CelestiaTestSuite) TestCelestiaAppMajorUpgrade() {
+	t := s.T()
+	if testing.Short() {
+		t.Skip("skipping major upgrade test in short mode")
+	}
+
+	ctx := context.Background()
+	const (
+		baseVersion   = "v4.0.7-mocha"
+		targetVersion = "d13d38a"
+		targetAppVer  = uint64(5) // The expected app_version after upgrade
+		chainID       = appconsts.TestChainID
+		homeDir       = "/var/cosmos-chain/celestia"
+	)
+
+	cfg := dockerchain.DefaultConfig(s.client, s.network).WithTag(baseVersion)
+
+	kr := cfg.Genesis.Keyring()
+
+	chain, err := dockerchain.NewCelestiaChainBuilder(t, cfg).Build(ctx)
+	s.Require().NoError(err)
+
+	err = chain.Start(ctx)
+	s.Require().NoError(err)
+
+	s.Require().NoError(wait.ForBlocks(ctx, 5, chain))
+
+	validatorNode := chain.GetNodes()[0]
+	rpcClient, err := validatorNode.GetRPCClient()
+	s.Require().NoError(err, "failed to get RPC client")
+
+	abciInfo, err := rpcClient.ABCIInfo(ctx)
+	s.Require().NoError(err, "failed to fetch ABCI info")
+	currentAppVer := abciInfo.Response.GetAppVersion()
+	s.T().Logf("Current app version before upgrade: %d", currentAppVer)
+
+	records, err := kr.List()
+	s.Require().NoError(err, "failed to list accounts")
+	s.Require().Len(records, len(chain.GetNodes()), "number of accounts does not match number of nodes")
+
+	// Signal for the upgrade to version 5 for each validator
+	for i, node := range chain.GetNodes() {
+		s.T().Logf("Signaling for upgrade to version %d from validator %d", targetAppVer, i)
+
+		hostname, err := node.GetInternalHostName(ctx)
+		s.Require().NoError(err, "failed to get internal hostname")
+
+		signalCmd := []string{
+			"celestia-appd", "tx", "signal", "signal", fmt.Sprintf("%d", targetAppVer),
+			"--home", homeDir,
+			"--from", records[i].Name,
+			"--keyring-backend", "test",
+			"--chain-id", chainID,
+			"--fees", "200000utia",
+			"--node", fmt.Sprintf("tcp://%s:26657", hostname),
+			"--yes",
+		}
+		stdout, stderr, err := node.Exec(ctx, signalCmd, nil)
+		s.Require().NoError(err, "failed to signal for upgrade: %s", stderr)
+		s.T().Logf("Signal output: %s", stdout)
+	}
+
+	s.Require().NoError(wait.ForBlocks(ctx, 2, chain))
+
+	hostname, err := validatorNode.GetInternalHostName(ctx)
+	s.Require().NoError(err, "failed to get internal hostname")
+
+	// Query the tally to see if we have enough voting power
+	tallyCmd := []string{
+		"celestia-appd", "query", "signal", "tally", fmt.Sprintf("%d", targetAppVer),
+		"--node", fmt.Sprintf("tcp://%s:26657", hostname),
+		"--output", "json",
+	}
+	tallyStdout, tallyStderr, err := validatorNode.Exec(ctx, tallyCmd, nil)
+	s.Require().NoError(err, "failed to query tally: %s", tallyStderr)
+	s.T().Logf("Tally output: %s", tallyStdout)
```

Comment:
this isn't used anywhere other than logging, do we need this?

Reply:
we used to have a validation to check if we have enough voting power. it might have been deleted by mistake